wire.md

```diff
@@ -0,0 +1,1008 @@
+
+#### 6.2.6 `post/join`
+
+Publicly announce membership in a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+
+`post_type` MUST be set to `4`.
+
+#### 6.2.7 `post/leave`
+
+Publicly announce termination of membership in a channel.
+
+field          | type               | desc
+---------------|--------------------|-------------------------------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+
+`post_type` MUST be set to `5`.
+
+### 6.3 Message Formats
+
+#### 6.3.1 Message Header
+
+All messages MUST begin with the following header fields:
+
+field         | type     | desc
+--------------|----------|-------------------------------------------------------------
+`msg_len`     | `varint` | number of bytes in rest of message, not including the `msg_len` field
+`msg_type`    | `varint` | a type identifier for the message, which controls which fields follow this header
+`circuit_id`  | `u8[4]`  | id of a circuit for an established path, or `[0,0,0,0]` for no circuit
+`req_id`      | `u8[4]`  | unique id of this request (random)
+
+Message-specific fields follow after the `req_id`.
+
+Each request and response type has a unique `msg_type` (see below), which
+controls which fields will immediately follow this header.
+
+Clients encountering a `msg_type` they do not know how to parse MUST ignore and
+discard it.
+
+The `circuit_id` field is not currently in use, and MUST be set to all zeros.
+
+The request ID, `req_id`, is a 32-bit number, generated randomly by the
+requester. It is used to uniquely identify the request during its lifetime
+across the peers who may handle it. For networks with up to 43 million active
+requests within a single cabal, the probability of collision remains below 1%.
+(i.e. `(1 - 1 / (2**32)) ** 43_000_000 > 0.99`)
+
+When generating a new `req_id`, a client MUST discard and generate a new one if
+the one generated collides with a known active `req_id`.
+
+When a client forwards a request to further peers, the `req_id` MUST NOT be
+changed, so that routing loops can be more easily detected by peers in the
+network.
+
+The protocol MAY be extended by implementers by creating additional
+`msg_type`s. Implementers MUST only use `msg_type > 255`. The
+first 256 are reserved for core protocol use.
+
+#### 6.3.2 Requests
+
+##### 6.3.2.1 Header
+
+Every request MUST have its header bytes be proceeded by the following request
+header fields:
+
+field      | type       | desc
+-----------|------------|-----------------------------------
+`ttl`      | `varint`   | number of network hops remaining
+
+The value of `ttl` MUST be between 0 and 16. 0 means "do not forward this
```

Comment:
Does using varint here add anything over saying it's one byte?

Reply:
It doesn't! Good catch.